internal/buildjob/buildjob_test.go

```diff
@@ -0,0 +1,235 @@
+package buildjob
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/dominodatalab/forge/api/forge/v1alpha1"
+	"github.com/dominodatalab/forge/internal/config"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	testK8sClient "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestBuildRegistryConfigs(t *testing.T) {
+	noAuthHost := "noauth-test.com"
+
+	noAuthRegistryConfig := v1alpha1.Registry{
+		Server: noAuthHost,
+		BasicAuth: v1alpha1.BasicAuthConfig{
+			Username: "",
+			Password: "",
+		},
+	}
+
+	noAuthRegistry := config.Registry{
+		Host:   noAuthHost,
+		NonSSL: false,
+	}
+
+	inlineUsername := "marge"
+	inlinePassword := "simpson"
+	inlineHost := "inline-test.com"
+
+	inlineRegistryConfig := v1alpha1.Registry{
+		Server: inlineHost,
+		BasicAuth: v1alpha1.BasicAuthConfig{
+			Username: inlineUsername,
+			Password: inlinePassword,
+		},
+	}
+
+	inlineRegistry := config.Registry{
+		Host:     inlineHost,
+		Username: inlineUsername,
+		Password: inlinePassword,
+		NonSSL:   false,
+	}
+
+	secretUsername1 := "johndoe"
+	secretPassword1 := "anothersecurepassword"
+	secretHost1 := "secret-test.com"
+
+	secretRegistry1 := config.Registry{
+		Host:     secretHost1,
+		Username: secretUsername1,
+		Password: secretPassword1,
+		NonSSL:   false,
+	}
+
+	secretUsername2 := "janedoe"
+	secretPassword2 := "yetanothersecurepassword"
+	secretHost2 := inlineHost // intentional conflict with inline auth host
+
+	secretRegistry2 := config.Registry{
+		Host:     secretHost2,
+		Username: secretUsername2,
+		Password: secretPassword2,
+		NonSSL:   false,
+	}
+
+	singleRegistrySecretName := "SecretWithSingleRegistry"
+	singleRegistrySecretNamespace := "SomeNamespace"
+	singleRegistrySecretContent := []byte(
+		fmt.Sprintf(`{"auths":{"%v":{"username":"%v","password":"%v"}}}`, secretHost1, secretUsername1, secretPassword1))
```

Comment:
a different way would have been to create a nested map and then use json.Encode to convert it.